### ISSUE TITLE
Temporary failure in name resolution; class=Net (12)

### ISSUE THREAD

**Issue by evanrichter** [ChrisGreenaway/cargo-local-registry#164]
Syncing fails during DNS lookup of github.com, although my DNS is working:

```bash
❯ cargo local-registry --sync ./Cargo.lock ../local_registry
    Updating crates.io index
warning: spurious network error (2 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
warning: spurious network error (1 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
error: failed to sync

Caused by:
  failed to load pkg lockfile

Caused by:
  failed to fetch `https://github.com/rust-lang/crates.io-index`

Caused by:
  failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
```

`dig` shows that my DNS is definitely working:

```
❯ dig github.com +short
140.82.113.3
```

I'm running this on WSL1, Debian buster

**Comment by ChrisGreenaway** [comment]
Thanks for the bug report.

This error comes from cargo. Perhaps https://github.com/rust-lang/cargo/issues/6513

If you run cargo build without the local registry (i.e. without the .cargo/config file), does that have the same error?

What kind of network infrastructure do you have? For example, are you inside a corporate network or have some network filtering / security software running?

Would you attach your Config.lock and Config.toml to the issue?

**Comment by evanrichter** [comment]
This is very strange, and unfortunately not reproducible at the moment. (It works now, though I changed nothing)

I'm just on a standard home network in the US.

At the time, only cargo local-registry was acting up. I could, for example run `cargo install bat` no problem. I don't have a ~/.cargo/config or any local config either.

I've attached my Cargo.lock and Cargo.toml for posterity (.txt added so github allowed the upload)

[Cargo.lock](https://github.com/ChrisGreenaway/cargo-local-registry/files/6127989/Cargo.lock.txt)
[Cargo.toml](https://github.com/ChrisGreenaway/cargo-local-registry/files/6127990/Cargo.toml.txt)
